test/sql/types/map/test_map_cardinality.test

```diff
@@ -0,0 +1,21 @@
+# name: test/sql/types/map/test_map_cardinality.test
+# description: Test cardinality function for maps
+# group: [map]
+
+statement error
+select cardinality(NULL)
+
+query I
+select CARDINALITY(MAP(LIST_VALUE(1, 2, 3, 4),LIST_VALUE(10, 9, 8, 7)))
+----
+4
+
+query I
+select CARDINALITY(MAP(LIST_VALUE(),LIST_VALUE()))
+----
+0
+
+query I
+select CARDINALITY(MAP())
```

Comment:
Similarly for CARDINALITY function:

* What about multiple rows?
* What about selection vectors?